<a href="https://colab.research.google.com/github/ivanhdiaz/EDA/blob/main/PROYECTO_FINAL_%E2%80%93_NOTEBOOK_ANAL%C3%8DTICO_CON_SQL_Y_PYTHON.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Sesión 1 Diploma**

# 1.- PORTADA Y DESCRIPCIÓN DEL PROYECTO

**PROYECTO FINAL – NOTEBOOK ANALÍTICO CON SQL Y PYTHON**

El proyecto tiene como fin aplicar las herramientas básicas aprendidas de Python a un caso de una empresa educativa de idiomas que ofrece una serie de programas tanto al público general como a empresas. Se cuenta con información histórica de gasto reales y presupuestadas de las distintas sedes y áreas de la organización desde el año 2024. El objetivo es determinar las sedes y áreas más eficientes en el manejo presupuestal de sus gastos desde dicho año.

Nombre Estudiante: Jerson Iván Huaytalla Díaz

Módulo: Introducción a Python y SQL

Título del análisis: Eficiencia del manejo presupuestal de gastos de una empresa del sector educativo desde el año 2024

Fuente del Dataset: Fuente propia

Objetivo del trabajo: Determinar las sedes y áreas más eficientes en el manejo presupuestal de sus gastos desde el año 2024.




# 2.- CARGA DEL DATASET

In [2]:
import pandas

In [4]:
datos = pandas.read_csv("/content/Dataset_Finanzas_Control_Gestion_2024_2026.csv")

In [5]:
datos

,Fecha,Año,Mes,Sede,Programa,Área,Centro_Costo,Cuenta_Contable,Tipo_Gasto,Proveedor,Presupuesto,Gasto_Real,Variación,Variación_%,Estado
0,2024-01-28,2024,Enero,Surco,Kids,Académica,CC1001,6349.0,Planilla,Cálidda,49000,53104,4104,8.38,Sobre presupuesto
1,2024-01-28,2024,Enero,Surco,Teens,Académica,CC1001,6346.0,Planilla,Cálidda,50000,58841,8841,17.68,Sobre presupuesto
2,2024-01-28,2024,Enero,Surco,Adults,Académica,CC1001,6373.0,Planilla,IBM,7500,7138,-362,-4.83,Dentro de presupuesto
3,2024-01-28,2024,Enero,Surco,Empresas,Académica,CC1001,6352.0,Planilla,Cálidda,43000,39119,-3881,-9.03,Dentro de presupuesto
4,2024-01-28,2024,Enero,Surco,Kids,Académica,CC1001,6369.0,Alquiler,Cálidda,5500,5978,478,8.69,Sobre presupuesto
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
46075,2026-12-28,2026,Diciembre,Los Olivos,Empresas,Comercial,CC1008,6340.0,Capacitación,IBM,36575,32441,-4134,-11.30,Dentro de presupuesto
46076,2026-12-28,2026,Diciembre,Los Olivos,Kids,Comercial,CC1008,6321.0,Seguridad,Office Depot,8550,9134,584,6.83,Sobre presupuesto
46077,2026-12-28,2026,Diciembre,Los Olivos,Teens,Comercial,CC1008,6381.0,Seguridad,Sodexo,3800,3716,-84,-2.21,Dentro de presupuesto
46078,2026-12-28,2026,Diciembre,Los Olivos,Adults,Comercial,CC1008,6310.0,Seguridad,Ricoh,52250,55209,2959,5.66,Sobre presupuesto


# 3.- EXPLORACIÓN INICIAL

In [11]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 46080 entries, 0 to 46079
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Fecha            46080 non-null  object 
 1   Año              46080 non-null  int64  
 2   Mes              46080 non-null  object 
 3   Sede             46080 non-null  object 
 4   Programa         46080 non-null  object 
 5   Área             46080 non-null  object 
 6   Centro_Costo     46080 non-null  object 
 7   Cuenta_Contable  45619 non-null  float64
 8   Tipo_Gasto       46080 non-null  object 
 9   Proveedor        45619 non-null  object 
 10  Presupuesto      46080 non-null  int64  
 11  Gasto_Real       46080 non-null  int64  
 12  Variación        46080 non-null  int64  
 13  Variación_%      46080 non-null  float64
 14  Estado           46080 non-null  object 
dtypes: float64(2), int64(4), object(9)
memory usage: 5.3+ MB


In [31]:
datos.head(5)

,Fecha,Año,Mes,Sede,Programa,Área,Centro_Costo,Cuenta_Contable,Tipo_Gasto,Proveedor,Presupuesto,Gasto_Real,Variación,Variación_%,Estado
0,2024-01-28,2024,Enero,Surco,Kids,Académica,CC1001,6349.0,Planilla,Cálidda,49000,53104,4104,8.38,Sobre presupuesto
1,2024-01-28,2024,Enero,Surco,Teens,Académica,CC1001,6346.0,Planilla,Cálidda,50000,58841,8841,17.68,Sobre presupuesto
2,2024-01-28,2024,Enero,Surco,Adults,Académica,CC1001,6373.0,Planilla,IBM,7500,7138,-362,-4.83,Dentro de presupuesto
3,2024-01-28,2024,Enero,Surco,Empresas,Académica,CC1001,6352.0,Planilla,Cálidda,43000,39119,-3881,-9.03,Dentro de presupuesto
4,2024-01-28,2024,Enero,Surco,Kids,Académica,CC1001,6369.0,Alquiler,Cálidda,5500,5978,478,8.69,Sobre presupuesto


In [34]:
datos.tail(5)

,Fecha,Año,Mes,Sede,Programa,Área,Centro_Costo,Cuenta_Contable,Tipo_Gasto,Proveedor,Presupuesto,Gasto_Real,Variación,Variación_%,Estado
46075,2026-12-28,2026,Diciembre,Los Olivos,Empresas,Comercial,CC1008,6340.0,Capacitación,IBM,36575,32441,-4134,-11.30,Dentro de presupuesto
46076,2026-12-28,2026,Diciembre,Los Olivos,Kids,Comercial,CC1008,6321.0,Seguridad,Office Depot,8550,9134,584,6.83,Sobre presupuesto
46077,2026-12-28,2026,Diciembre,Los Olivos,Teens,Comercial,CC1008,6381.0,Seguridad,Sodexo,3800,3716,-84,-2.21,Dentro de presupuesto
46078,2026-12-28,2026,Diciembre,Los Olivos,Adults,Comercial,CC1008,6310.0,Seguridad,Ricoh,52250,55209,2959,5.66,Sobre presupuesto
46079,2026-12-28,2026,Diciembre,Los Olivos,Empresas,Comercial,CC1008,6338.0,Seguridad,Sodexo,37050,42460,5410,14.60,Sobre presupuesto


# 4.- ANÁLISIS DE CALIDAD DE DATOS

In [37]:
datos.isnull()

,Fecha,Año,Mes,Sede,Programa,Área,Centro_Costo,Cuenta_Contable,Tipo_Gasto,Proveedor,Presupuesto,Gasto_Real,Variación,Variación_%,Estado
0,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
46075,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
46076,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
46077,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
46078,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False


In [38]:
datos.isnull().sum()

,0
Fecha,0
Año,0
Mes,0
Sede,0
Programa,0
Área,0
Centro_Costo,0
Cuenta_Contable,461
Tipo_Gasto,0
Proveedor,461


In [40]:
datos.duplicated()

,0
0,False
1,False
2,False
3,False
4,False
...,...
46075,False
46076,False
46077,False
46078,False


In [42]:
datos.fillna(0)

,Fecha,Año,Mes,Sede,Programa,Área,Centro_Costo,Cuenta_Contable,Tipo_Gasto,Proveedor,Presupuesto,Gasto_Real,Variación,Variación_%,Estado
0,2024-01-28,2024,Enero,Surco,Kids,Académica,CC1001,6349.0,Planilla,Cálidda,49000,53104,4104,8.38,Sobre presupuesto
1,2024-01-28,2024,Enero,Surco,Teens,Académica,CC1001,6346.0,Planilla,Cálidda,50000,58841,8841,17.68,Sobre presupuesto
2,2024-01-28,2024,Enero,Surco,Adults,Académica,CC1001,6373.0,Planilla,IBM,7500,7138,-362,-4.83,Dentro de presupuesto
3,2024-01-28,2024,Enero,Surco,Empresas,Académica,CC1001,6352.0,Planilla,Cálidda,43000,39119,-3881,-9.03,Dentro de presupuesto
4,2024-01-28,2024,Enero,Surco,Kids,Académica,CC1001,6369.0,Alquiler,Cálidda,5500,5978,478,8.69,Sobre presupuesto
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
46075,2026-12-28,2026,Diciembre,Los Olivos,Empresas,Comercial,CC1008,6340.0,Capacitación,IBM,36575,32441,-4134,-11.30,Dentro de presupuesto
46076,2026-12-28,2026,Diciembre,Los Olivos,Kids,Comercial,CC1008,6321.0,Seguridad,Office Depot,8550,9134,584,6.83,Sobre presupuesto
46077,2026-12-28,2026,Diciembre,Los Olivos,Teens,Comercial,CC1008,6381.0,Seguridad,Sodexo,3800,3716,-84,-2.21,Dentro de presupuesto
46078,2026-12-28,2026,Diciembre,Los Olivos,Adults,Comercial,CC1008,6310.0,Seguridad,Ricoh,52250,55209,2959,5.66,Sobre presupuesto


# 5.- ESTADÍSTICA DESCRIPTIVA

In [19]:
datos["Sede"].value_counts()

,count
Sede,
Surco,9216
Miraflores,9216
San Miguel,9216
Lima Centro,9216
Los Olivos,9216


In [20]:
datos["Área"].value_counts()

,count
Área,
Académica,5760
Marketing,5760
Administración,5760
Finanzas,5760
TI,5760
RRHH,5760
Operaciones,5760
Comercial,5760


In [15]:
datos.describe()

,Año,Cuenta_Contable,Presupuesto,Gasto_Real,Variación,Variación_%
count,46080.000000,45619.000000,46080.000000,46080.000000,46080.000000,46080.000000
mean,2025.000000,6354.507091,32644.870117,33619.344705,974.474588,2.991350
std,0.816505,26.042025,16825.308096,17617.523500,3231.947354,8.676616
min,2024.000000,6310.000000,3800.000000,3359.000000,-8208.000000,-12.000000
25%,2024.000000,6332.000000,18050.000000,18614.500000,-1029.000000,-4.562500
50%,2025.000000,6354.000000,32500.000000,33125.000000,615.000000,2.980000
75%,2026.000000,6377.000000,47000.000000,47837.500000,2848.000000,10.550000
max,2026.000000,6399.000000,68425.000000,80633.000000,12208.000000,18.000000


 - El presupuesto promedio es de 32,644.87
 - El gasto real promedio es de 33,619.34
 - La var. presupuestal promedio es de 2.99% y la máxima es de 18%
 - El gasto real máximo es de 80,633


In [55]:
datos.where(datos["Gasto_Real"]>68425)

,Fecha,Año,Mes,Sede,Programa,Área,Centro_Costo,Cuenta_Contable,Tipo_Gasto,Proveedor,Presupuesto,Gasto_Real,Variación,Variación_%,Estado
0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
46075,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
46076,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
46077,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
46078,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [21]:
datos["Tipo_Gasto"].value_counts()

,count
Tipo_Gasto,
Planilla,5760
Alquiler,5760
Servicios,5760
Marketing,5760
Tecnología,5760
Mantenimiento,5760
Capacitación,5760
Seguridad,5760


In [16]:
datos.describe(include="object")

,Fecha,Mes,Sede,Programa,Área,Centro_Costo,Tipo_Gasto,Proveedor,Estado
count,46080,46080,46080,46080,46080,46080,46080,45619,46080
unique,36,12,5,4,8,8,8,12,2
top,2024-01-28,Enero,Surco,Kids,Académica,CC1001,Planilla,Cálidda,Sobre presupuesto
freq,1280,3840,9216,11520,5760,5760,5760,3951,27584


 - Se cuenta con 5 sedes
 - Se cuenta con 8 áreas
 - Se cuenta con 8 tipos de gastos diferentes
 - La sede Surco es la que predomina
 - El área académica es la que predomina
 - El gasto de planilla es el que predomina

In [22]:
datos["Proveedor"].value_counts()

,count
Proveedor,
Cálidda,3951
Microsoft,3932
Office Depot,3905
Movistar,3857
Entel,3804
Claro,3792
Ricoh,3757
IBM,3755
Luz del Sur,3740


In [23]:
datos["Estado"].value_counts()

,count
Estado,
Sobre presupuesto,27584
Dentro de presupuesto,18496


In [59]:
datos.groupby(["Año","Estado"])["Variación"].sum()

Año   Estado               
2024  Dentro de presupuesto   -11775083
      Sobre presupuesto        27144684
2025  Dentro de presupuesto   -12193147
      Sobre presupuesto        27243796
2026  Dentro de presupuesto   -12423450
      Sobre presupuesto        26906989
Name: Variación, dtype: int64

In [51]:
datos.groupby(["Año","Área"])["Variación"].sum()

Año   Área          
2024  Académica         2078980
      Administración    2020048
      Comercial         1839503
      Finanzas          1837951
      Marketing         2025902
      Operaciones       1820892
      RRHH              1859950
      TI                1886375
2025  Académica         2105378
      Administración    1942437
      Comercial         1698419
      Finanzas          1865619
      Marketing         2020505
      Operaciones       1844382
      RRHH              1852727
      TI                1721182
2026  Académica         1835058
      Administración    1958069
      Comercial         1971824
      Finanzas          1779597
      Marketing         1567676
      Operaciones       1861101
      RRHH              1845659
      TI                1664555
Name: Variación, dtype: int64

# 6.- VISUALIZACIONES CON PLOTLY

In [60]:
import pandas as pd

import plotly.express as px

import numpy as np

In [ ]:
fig = px.bar(x=Año, y=Variación, title="Variación vs Ppto por año",
             labels={'x': 'Productos', 'y': 'Unidades Vendidas'})